### PR TITLE
CI VMs: bump to f39 + f38

### PR DESCRIPTION
**Pull request by edsantiago**
...from f38 + f37.

Requires one minor e2e test change, to handle an error logging
change in conmon 2.1.8.

Also, this is important, requires crun-1.9.1 because of a kernel
symlink change; see https://github.com/containers/crun/pull/1309
The VM images here were carefully built to include that. By the
time the next VM images get built, it should be default.

Signed-off-by: Ed Santiago <santiago@redhat.com>
```release-note
None
```

### CODE REVIEW

**Review by vrothberg — APPROVED** [review]
Changes LGTM

Restarted the two jobs which looked like flakes

**Comment by vrothberg** [comment]
Windows Cross failure seems to persist but I do not understand it

**Comment by Luap99** [comment]
> This version of pandoc has been compiled without Lua support.

Looks like the pandoc package was split into several packages https://hackage.haskell.org/package/pandoc-3.0/changelog

Seems to be already reported against the fedora package: https://bugzilla.redhat.com/show_bug.cgi?id=2238149
I guess to unblock this PR it would be possible to keep running this windows cross job on f38?

**Comment by edsantiago** [comment]
Thank you for finding that, @Luap99. I saw "Windows" and my brain froze up. 

Unless there is an urgent need to start testing on f39, I'm ok with waiting for a fixed pandoc. My VM-build PR is https://github.com/containers/automation_images/pull/302 and it would be nice (IMO) to remove the special-case `crun` kludge by waiting a few days.

**Comment by edsantiago** [comment]
>would you consider including [the passthrough_envars() updates](https://github.com/containers/podman/pull/19751/commits/f4331df34eab393d5cf5c99be6439294da0c5570#diff-87b53ab90b08478b1624b8f218f6d0b6eab834f2cf81dd73581540c3a0dfb63c)

Done (on my local branch). 

>Also [Paul's commit can be cherry-picked here](https://github.com/containers/podman/pull/19751/commits/ef8d88cb5a896aaf9adca8edbea23355cd88a012) safely.

Done, ibid.

I am monitoring the pandoc BZ, also periodically checking bodhi for pandoc builds. If I see a build, I will rerun https://github.com/containers/automation_images/pull/302 . If there's nothing by Monday, I will look into some other way to fix the Windows problem.

**Comment by ashley-cui** [comment]
> IIRC, the output from that job is consumed by the other windows tasks. I believe @ashley-cui is working on native builds for Windows, which can/should probably replace the cross-build. Is that you're eventual goal Ashley?

Yeah, eventual builds and tests should run on native Windows

**Comment by cevich** [comment]
> Yeah, eventual builds and tests should run on native Windows

Great, you can get that implemented before Monday, right :wink: (kidding).

**Comment by TomSweeneyRedHat** [comment]
Changes LGTM 
I'm not sure what to do given the Windows Cross failure.

**Comment by edsantiago** [comment]
Should this pass CI, here's what's important to review:
* First and foremost, the CI-VM PR: https://github.com/containers/automation_images/pull/302 
  * It brings in `crun-1.9.2-1` in a dirty way. This is an ends-justify-means kludge IMO. If you disagree, reject.
  * Even more so for `pandoc`, bringing in a scratch build of pandoc-with-lua in order to pass the Windows test
  * the rest of that PR should be noncontroversial. It might help to review as two commits (one of them is just `s/egrep/grep -E/` because egrep is deprecated).
* If the CI-VM PR looks good, then review this one:
  * The first (of three) commits is mine. I think it should be noncontroversial.
  * Second and third are brought in from #19751 at @cevich's request

As of this writing, Windows has passed (yay) which means pandoc works. I will be asleep long before any int/sys jobs complete.

File: .cirrus.yml
Comment (cevich) [inline .cirrus.yml:29]: WDYT is it worth classifying this as `fedora-39 (beta)` or some-such?

I believe it will still [pass the distro-verison check](https://github.com/containers/podman/pull/20162/files#diff-f3866c83e5fdf3650df2ad5b9c0c18efaa71891d23675732cc931e90fd2b10faR66-R69) and should display in task-names as a reminder.

I'm not going to block over this, since it could be done as a follow-up.  I'm feel it may be an important visual reminder that somebody needs to do another F39 build whenever it's released.

**Review by cevich — COMMENTED** [review]
LGTM with one small thing for consideration.

File: .cirrus.yml
Comment (edsantiago) [reply .cirrus.yml:29]: Ooh, I like that. I'd prefer to do it as a followup, because I don't know how/if it'd affect logformatter URLs.

Comment (edsantiago) [reply .cirrus.yml]: Seems to work! In [urls](https://api.cirrus-ci.com/v1/artifact/task/5308206646296576/html/sys-remote-fedora-39%CE%B2-root-host-sqlite.log.html) and in github display: 
beta

Re-pushed with β

**Review by cevich — APPROVED** [review]
LGTM assuming happy tests.  Thanks Ed.

**Comment by edsantiago** [comment]
Tests are green again. @containers/podman-maintainers are we ready for f39?

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Pull request merged**
